pkg/controller/controller.go

```diff
@@ -271,6 +274,8 @@ shutdown:
 		c.getLogger().WithField(fieldConsecutiveErrors, errorRetries).
 			WithError(err).Warn("Error on Controller stop")
 	}
+
+	close(c.terminated)
 }
 
 func (c *Controller) stopController() {
```

Comment:
I can't think of a way it would be a problem, but stopController doesn't look at `terminated` at all. I guess it's a no-op, ultimately.

Reply:
`stopController()` signal the controller to quit after which `runController()` breaks out of the loop and closes `terminated`. The `terminated` is always created before `runController()` is started and is always closed when `runController()` terminates.